pkg/workflows/wasm/host/module.go

```diff
@@ -580,7 +587,49 @@ func runWasm[I idMessage, O proto.Message](
 }
 
 func (m *module) SetCapabilityExecutor(handler CapabilityExecutor) error {
-	return errors.New("not implemented")
+	callCapAsync := m.cfg.CallCapability
+	if handler != nil {
+		callCapAsync = toCallCapAsync(m.capabilityCallStore, handler)
+	}
+
+	// shadow any existing implementation with the one passed by the call
+	if err := m.linker.FuncWrap(
+		"env",
+		"call_capability",
+		createCallCapFn(m.cfg.Logger, m.executeStore, callCapAsync),
+	); err != nil {
+		return fmt.Errorf("error wrapping callcap func: %w", err)
+	}
+
+	return nil
+}
+
+func toCallCapAsync(
+	capStore *store[<-chan *sdkpb.CapabilityResponse],
+	handler CapabilityExecutor,
+) func(ctx context.Context, req *sdkpb.CapabilityRequest) ([sdk.IdLen]byte, error) {
+	return func(ctx context.Context, req *sdkpb.CapabilityRequest) ([sdk.IdLen]byte, error) {
+		callId := uuid.NewString()
+		var idBuffer [sdk.IdLen]byte
+		copy(idBuffer[:], []byte(callId))
+
+		go func() {
+			ch := make(chan *sdkpb.CapabilityResponse, 1)
+			resp, err := handler.CallCapability(ctx, req)
+			if err != nil {
+				resp = &sdkpb.CapabilityResponse{
+					Response: &sdkpb.CapabilityResponse_Error{
+						Error: err.Error(),
+					},
+				}
+			}
+			// TODO: Delete id on await
+			ch <- resp
```

Comment:
You should select  in case the context is cancelled. For example, if the executable stops running, we don't need to be blocked anymore.